kolibri/core/assets/src/components_demo.js

```diff
@@ -46,67 +51,11 @@ var ComponentDemoPlugin = KolibriModule.extend({
-    /*
-    The template and regions attributes define the DOM containers for subviews.
-    Regions are a Marionette abstraction which handle View loading/destroying -- they are essentially named DOM
-    elements.
-     */
-    // The component Views provide semantic classes -- in the template, we use classes only to identify regions
-    // In other words, region classes shouldn't be the target of style rules
-    template: _.template('<div class="region1"></div>' +
-                         '<div class="region2"></div>'),
-
-    tagName: 'div',
-
-    className: 'tiwtd',
-
-    regions: {
-        textinput: '.region1',  // The selector refers to DOM elements in the View's template only.
-        taglist: '.region2'
-    },
-
-    /*
-    A good ol' fashioned initialize function. Setup the child views, but *don't* display them. That's
-    the responsibility of the framework presently, and in the future our virtual dom(?)
-     */
-    initialize: function() {
-        // A view should pass on it's own model or some element in it. Then events triggered on the model can
-        // be handled by the Application.
-        this.textInputField = new components.TextLineInput({model: this.model});
-        this.tagList = new components.TagList({collection: this.model.get('tags')});
-
-        // The list of events of base elements emit (and the arguments passed with those events)
-        // should be enumerated in the documentation.
-        // In this case, I'm assuming that TagList base component emits a 'tag_clicked' event, namespaced by 'tag_list'
-        // We may choose a different convention.
-        this.listenTo(this.tagList, 'tag_list:tag_clicked', function(tag_name) {
-            console.log(tag_name + ' was clicked!');
-            // Don't try to redraw -- just change the underlying model and let the app handle redrawing.
-            // Note that Marionette handles the redraw itself, presumably in the CollectionView's logic.
-            // If we want to override this behavior, we'll have to look into the internals.
-            var coll = this.model.get('tags');
-            var tag = coll.findWhere({name: tag_name});
-            coll.remove(tag);
-        });
-        // Similarly, I'm assuming that textInputFields emit the following event+arguments.
-        this.listenTo(this.textInputField, 'text_input:text_changed', function(text){
-            this.model.get('tags').push(new Backbone.Model({
-                name: text
-            }));
-            // Does this count as "introducing state"???
-            this.textInputField.clear();
-        });
-    },
 
-    // This convention is recommended by Marionette for efficient repaints.
-    // See: http://marionettejs.com/docs/v2.4.5/marionette.layoutview.html#efficient-nested-view-structures
-    onBeforeShow: function() {
-        this.showChildView('taglist', this.tagList);
-        this.showChildView('textinput', this.textInputField);
+var User = Backbone.Model.extend({
```

Comment:
No. But you could define a [validate](http://backbonejs.org/#Model-validate) method on Models and raise exceptions to avoid arbitrary attributes from being set & retrieved.